TST: test_acorr (replaced image comparison with figure comparion)

pharshalp opened the pull request:
## PR Summary
attempting to address the first bullet from #15054 🤞

~~**Old image for acorr**
~~https://user-images.githubusercontent.com/15175620/73325631-3a6e1880-421d-11ea-9d22-c4939741edd5.png~~
~~**New image for acorr**
https://user-images.githubusercontent.com/15175620/73325639-435eea00-421d-11ea-9542-605faf629ca7.png~~

~~**Old image for image_interps**
https://user-images.githubusercontent.com/15175620/73325664-58d41400-421d-11ea-9c27-4408662f7519.png~~

~~**New image for image_interps**
https://user-images.githubusercontent.com/15175620/73325673-5f628b80-421d-11ea-8f8d-347783f169a0.png~~

## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

<!--
Thank you so much for your PR!  To help us review your contribution, please
consider the following points:

- A development guide is available at https://matplotlib.org/devdocs/devel/index.html.

- Help with git and github is available at
  https://matplotlib.org/devel/gitwash/development_workflow.html.

- Do not create the PR out of master, but out of a separate branch.

- The PR title should summarize the changes, for example "Raise ValueError on
  non-numeric input to set_xlim".  Avoid non-descriptive titles such as
  "Addresses issue #8576".

- The summary should provide at least 1-2 sentences describing the pull request
  in detail (Why is this change required?  What problem does it solve?) and
  link to any relevant issues.

- If you are contributing fixes to docstrings, please pay attention to
  http://matplotlib.org/devel/documenting_mpl.html#formatting.  In particular,
  note the difference between using single backquotes, double backquotes, and
  asterisks in the markup.

We understand that PRs can sometimes be overwhelming, especially as the
reviews start coming in.  Please let us know if the reviews are unclear or
the recommended next step seems overly demanding, if you would like help in
addressing a reviewer's comments, or if you have been waiting too long to hear
back on your PR.
-->

pharshalp commented:
realized that svg wasn't generated automatically... will update it soon.

anntzer commented:
As per https://github.com/matplotlib/matplotlib/pull/16363#issuecomment-579710927 I think the second image should be left as is.  The first one could be turned into a `@check_figures_equal` between calling ax.acorr and manually calling np.correlate and generating the corresponding plot using ax.vlines/ax.hline (which is what ax.acorr uses internally).

pharshalp commented:
> 
> 
> As per [#16363 (comment)](https://github.com/matplotlib/matplotlib/pull/16363#issuecomment-579710927) I think the second image should be left as is, sorry again. The first one could be turned into a `@check_figures_equal` between calling ax.acorr and manually calling np.correlate and generating the corresponding plot using ax.vlines/ax.hline (which is what ax.acorr uses internally).

updated the tests as suggested (restored the images and `image_interps.py`, replaced image comparison for `ax.acorr` with `@check_figures_equal`).
